### DISCUSSION TITLE
Add special characters.

### DISCUSSION THREAD

**Discussion by duedipicche** [Motion-Project/motion#1509] [General]
Request to add new special characters "°" (degrees) and "*" (asterisk) to the options text_left, text_right to implement gps, temperatures etc  to the overlaid text.

**Comment by tosiara** [comment]
I can confirm that '*' (ascii hex 0x2a) and '°' (unicode hex 0xc2 0xb0) does not render:
```
text_left |*|-°-|
```
![image](https://user-images.githubusercontent.com/3600973/97083883-ee20ff00-161b-11eb-98ca-eeff56b8c4d5.png)

**Comment by romgen** [comment]
This is related to issue #1082 .
You can see in [https://github.com/Motion-Project/motion/blob/master/src/draw.c](url) that the currect code to include a text overlay does only supoprt ascii characters < 128 (therefore excluding "°") and on the other side, it does only support few special ascii chars below 128, this is the list of suported special characters for the text overlay:
"/()\_+-:.,|<>#~@

If you want more, motion should use afont rendering engine like font-rs or freetype. For a quick-fix, if you're up to it, you could include your missing character in the draw.c (begining on line 43) and send a pull request :-)

**Comment by tosiara** [comment]
Adding a new symbol is just a matter of such a change:
```
diff --git a/src/draw.c b/src/draw.c
index aee2224..786a94a 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -182,6 +182,19 @@ struct draw_char draw_table[]= {
             {0,0,1,1,1,0,0}
         }
     },
+    {
+        '%',
+        {
+            {0,2,2,0,0,2,1},
+            {2,1,1,2,2,1,1},
+            {0,2,2,2,1,1,2},
+            {0,0,2,1,1,2,0},
+            {0,2,1,1,2,0,0},
+            {2,1,1,2,2,2,0},
+            {1,1,2,2,1,1,2},
+            {1,2,0,0,2,2,0}
+        }
+    },
     {
         '"',
         {
```
We only need to carefully draw more symbols. This should not break an existing functionality
